src/common/perf_counters.cc

```diff
@@ -135,7 +135,12 @@ void PerfCountersCollectionImpl::dump_formatted_generic(
     const std::string &counter) const
 {
   f->open_object_section("perfcounter_collection");
-  
+  // close out all of counters collection immediately if collection is empty
+  if (m_loggers.empty()) {
```

Comment:
Why is this needed?  We are already handling the empty case in the labeled branch and the unlabeled branch shouldn't need any special treatment.